diff --git a/src/corelib/Qt6WasmMacros.cmake b/src/corelib/Qt6WasmMacros.cmake
--- a/src/corelib/Qt6WasmMacros.cmake
+++ b/src/corelib/Qt6WasmMacros.cmake
@@ -4 +4,3 @@
+include(QtPublicWasmToolchainHelpers)
+
 # Copy in Qt HTML/JS launch files for apps.
diff --git a/cmake/QtPublicWasmToolchainHelpers.cmake b/cmake/QtPublicWasmToolchainHelpers.cmake
--- a/cmake/QtPublicWasmToolchainHelpers.cmake
+++ b/cmake/QtPublicWasmToolchainHelpers.cmake
@@ -77,10 +77,14 @@ function(__qt_internal_get_qt_build_emsdk_version out_var)
     if(QT6_INSTALL_PREFIX)
         set(WASM_BUILD_DIR "${QT6_INSTALL_PREFIX}")
     elseif(QT_BUILD_DIR)
         set(WASM_BUILD_DIR "${QT_BUILD_DIR}")
     endif()
     if(EXISTS "${WASM_BUILD_DIR}/src/corelib/global/qconfig.h")
         file(READ "${WASM_BUILD_DIR}/src/corelib/global/qconfig.h" ver)
+    elseif(QT6_INSTALL_PREFIX AND QT6_INSTALL_HEADERS
+            AND EXISTS "${QT6_INSTALL_PREFIX}/${QT6_INSTALL_HEADERS}/QtCore/qconfig.h")
+        file(READ
+            "${QT6_INSTALL_PREFIX}/${QT6_INSTALL_HEADERS}/QtCore/qconfig.h" ver)
     elseif(EXISTS "${WASM_BUILD_DIR}/include/QtCore/qconfig.h")
         file(READ "${WASM_BUILD_DIR}/include/QtCore/qconfig.h" ver)
     endif()
